src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs

```diff
@@ -101,43 +109,93 @@ internal CloudTable GetDiagnosticEventsTable(DateTime? now = null)
             return _diagnosticEventsTable;
         }
 
-        private static string GetCurrentTableName(DateTime now)
+        private static string GetTableName(DateTime date)
         {
-            return $"{TableNamePrefix}{now:yyyyMM}";
+            return $"{TableNamePrefix}{date:yyyyMM}";
         }
 
         protected internal virtual async void OnFlushLogs(object state)
         {
             await FlushLogs();
         }
 
+        private async Task PurgePreviousEventVersions()
+        {
+            _logger.LogTrace("Purging previous diagnostic event versions.");
+
+            try
+            {
+                var tables = await TableStorageHelpers.ListTablesAsync(TableClient, TableNamePrefix);
+
+                foreach (var table in tables)
+                {
+                    var tableRecords = await table.ExecuteQuerySegmentedAsync(new TableQuery<DiagnosticEvent>(), null);
+
+                    // Skip tables that have 0 records
+                    if (tableRecords.Results.Count == 0)
+                    {
+                        continue;
+                    }
+
+                    // Delete table if it doesn't have records with EventVersion
+                    var eventVersionDoesNotExists = tableRecords.Results.Any(record => string.IsNullOrEmpty(record.EventVersion) == true);
+                    if (eventVersionDoesNotExists)
+                    {
+                        _logger.LogTrace("Deleting table '{tableName}' as it contains records without an EventVersion.", table.Name);
+                        await table.DeleteIfExistsAsync();
+                        continue;
+                    }
+
+                    // If the table does have EventVersion, query if it is an outdated version
+                    var eventVersionOutdated = tableRecords.Results.Any(record => record.EventVersion.CompareTo(DiagnosticEvent.CurrentEventVersion) < 0);
+                    if (eventVersionOutdated)
+                    {
+                        _logger.LogTrace("Deleting table '{tableName}' as it contains records with an outdated EventVersion.", table.Name);
+                        await table.DeleteIfExistsAsync();
+                    }
+                }
+
+                _purged = true;
+            }
+            catch (Exception ex)
```

Comment:
Should we also wrap this code inside a retry snippet so that transient issues (connection dropped to table storage) has minimal impact?